src/engraving/libmscore/beam.h

```diff
@@ -33,18 +33,105 @@ class ChordRest;
 class Factory;
 class Skyline;
 class System;
-
+class Beam;
 enum class ActionIconType;
 enum class SpannerSegmentType;
 
-struct BeamFragment;
+//---------------------------------------------------------
+//   BeamFragment
+//    position of primary beam
+//    idx 0 - DirectionV::AUTO or DirectionV::DOWN
+//        1 - DirectionV::UP
+//---------------------------------------------------------
+
+struct BeamFragment {
+    double py1[2];
+    double py2[2];
+};
 
 struct BeamSegment {
     mu::LineF line;
     int level;
     bool above; // above level 0 or below? (meaningless for level 0)
 };
 
+struct TremAnchor {
+    ChordRest* chord1;
+    double y1;
+    double y2;
+};
+
+class BeamLayout
+{
+private:
+    static constexpr std::array _maxSlopes = { 0, 1, 2, 3, 4, 5, 6, 7 };
+    enum class BeamType {
+        INVALID,
+        BEAM,
+        TREMOLO
+    };
+    BeamType _beamType{ BeamType::INVALID };
+    EngravingItem* _e{ nullptr };
+    Beam* _beam{ nullptr };
```

Comment:
All the class members should have the m_ prefix

Reply:
done!